### ISSUE TITLE
distinct() after joining data.table's

### ISSUE THREAD

**Issue by wtcooper** [hadley/dplyr#1285]
Running distinct() after joining two data.table's leads to a data set with just unique keys (the join variable), not unique rows.  Should this be changed to be comparable with distinct() if running on underlying data.frame?  

e.g.
 
```r
x = data.table(id=sample(1:3, 10, replace=T), xval=rnorm(10))
y = data.table(id=sample(1:3, 10, replace=T), yval=rnorm(10))

j = x %>% left_join(y, by="id")
jDT = j %>% distinct()                                  # only 3 obs - unique keys
jDF = j %>% data.frame() %>% distinct()     # expected unique rows
```

**Comment by romainfrancois** [comment]
That's because: 

```r
> key(j)
[1] "id"
```

probably because of `inner_join.data.table` impl: 

```
> inner_join.data.table
function (x, y, by = NULL, copy = FALSE, ...)
{
    by <- common_by(by, x, y)
    if (!identical(by$x, by$y)) {
        stop("Data table joins must be on same key", call. = FALSE)
    }
    y <- auto_copy(x, y, copy = copy)
    x <- data.table::copy(x)
    y <- data.table::copy(y)
    data.table::setkeyv(x, by$x)
    data.table::setkeyv(y, by$x)
    out <- merge(x, y, by = by$x, allow.cartesian = TRUE)
    grouped_dt(out, groups(x))
}
<environment: namespace:dplyr>
```

**Comment by wtcooper** [comment]
I see already addressed in #990 and referenced in #1208, thanks.
